src/client/pkg/config/config.go

```diff
@@ -10,8 +10,17 @@ import (
 	"github.com/satori/go.uuid"
 )
 
-var configDirPath = filepath.Join(os.Getenv("HOME"), ".pachyderm")
-var configPath = filepath.Join(configDirPath, "config.json")
+const configEnvVar = "PACH_CONFIG"
+
+var defaultConfigDir = filepath.Join(os.Getenv("HOME"), ".pachyderm")
+var defaultConfigPath = filepath.Join(defaultConfigDir, "config.json")
+
+func configPath() string {
+	if env, ok := os.LookupEnv(configEnvVar); ok {
+		return env
+	}
+	return defaultConfigPath
+}
```

Comment:
Might as well make this an env struct like our other packages.

Reply:
I want to push back on this a bit because I think it would make the code quite a bit more complicated, and I want to make sure it makes sense.

These config functions are only called in two places: the client library and the metrics library, so to use an env struct, we'd have to be using it somewhere else and then pass the value of this environment variable into those libraries, which would pass it here. In particular, we'd have to create one in pachctl (which doesn't currently use one), and then everywhere that creates a client library (or sends cluster metrics) would have to be updated to pass a config path, and everywhere a client is created in pachd, this field would be empty, but every library used by pachctl that creates a client would also have to accept a config path so that it could wire the value through to the client library, and then all of those call sites would have to be updated too, etc.

As it is, I think this is actually cleaner (and it resembles other environment variables that the client library checks, like `ADDRESS` and `PACH_CA_CERTS`), and it's certainly less work. Even if we do want to change this, would it make sense to do it after this PR is submitted?

Reply:
I don't think you'd have to initialize the env struct somewhere else. You can read env vars into a struct either by calling `cmdutil.Main(mainFunc, envStruct)` or by just `cmdutil.Populate(envStruct)` which you should be able to do from anywhere. I think it'll probably ultimately be less code to do it this way.

That being said, this is definitely not a big deal so if it seems like it's going to be a real amount of work or slow you down feel free to not do it.